lib/cstruct.ml

```diff
@@ -410,16 +410,16 @@ let rec fold f next acc = match next () with
 
 let append cs1 cs2 =
   let l1 = len cs1 and l2 = len cs2 in
-  let cs = create (l1 + l2) in
+  let cs = create_unsafe (l1 + l2) in
   blit cs1 0 cs 0  l1 ;
   blit cs2 0 cs l1 l2 ;
   cs
 
 let concat = function
-  | []   -> create 0
+  | []   -> create_unsafe 0
```

Comment:
i've opened a separate issue for this in https://github.com/mirage/ocaml-cstruct/issues/163